projects/core/src/occ/adapters/cms/legacy-occ-cms-component.adapter.ts

```diff
@@ -44,7 +44,7 @@ export class LegacyOccCmsComponentAdapter extends OccCmsComponentAdapter {
 
     return this.http
       .post<Occ.ComponentList>(
-        this.getComponentsEndpoint(requestParams, fields),
+        this.getComponentsEndpoint(requestParams, fields, ''),
```

Comment:
We are passing empty string as a third argument (userId). Shouldn't be here any default value like: 'anonymous' or 'current'? Or maybe we can make this argument as an optional?

Reply:
Tough question. This is the legacy adapter. The old backend will not be compatible with the new endpoint. If I add the default value, the current adapter will add userId params. Making this value optional would be incorrect in my opinion because with the new API we need it. Now I don't know the use case of this legacy adapter so it's hard for me to say what to do with it. With this change the users have to be prepared to use old endpoint.

Reply:
removed it and made userId optional